zipkin-storage/cassandra-v1/src/main/java/zipkin2/storage/cassandra/v1/IndexTraceIdByServiceName.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2015-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.storage.cassandra.v1;
+
+import com.datastax.driver.core.BoundStatement;
+import com.datastax.driver.core.querybuilder.Insert;
+import com.datastax.driver.core.querybuilder.QueryBuilder;
+import java.util.concurrent.ThreadLocalRandom;
+
+// QueryRequest.serviceName
+final class IndexTraceIdByServiceName extends IndexTraceId.Factory {
+  // Legacy behaviour was to use a static singleton of ThreadLocalRandom for bucket selection
```

Comment:
Do we need to preserve this?

Reply:
I punted to https://github.com/openzipkin/zipkin/pull/3137#discussion_r449952865 in case I'm missing something. let's carry on here?